### PR TITLE
Adds more network diagnostics to the multitool

### PR DESCRIPTION
**Pull request by penny64**
[QOL]
## About the PR 

The multitool can now be smacked against computers, fabs, PDAs, bots, basically anything to get the netid and/or frequency of that device; as well as similar information like the control and beacon frequencies of bots

image

## Why's this needed? 

This is important for several reasons, it is often difficult to work with network devices because it is not possible to find their netid.
With this player are now able to actually install a new databank and mount it without being completely blind. A new player can find that an airlock uses the frequency 1411.
The frequency output adds a ton of in game documentation, allowing players to research and learn about wireless devices without checking the wiki(if your chosen device is even documented).
Features and diagnostic information and documentation is good.

## Changelog

```changelog
(u)Penny
(*)Multitools now show the netid and wifi frequency of devices they're used on
```

### CODE REVIEW

**Comment by witheld9** [comment]
Alright that looks good, I've run around smacking everything with the multitool and I think the GPS was the only weird thing left!

File: code/obj/item/tool/multitool.dm [hunk 1]
```diff
@@ -26,3 +26,59 @@
...
+
+	if(net_id)
+		boutput(user, "<span class='alert'>NETID#[net_id]</span>")
+	if(frequency)
+		boutput(user, "<span class='alert'>FREQ#[frequency]</span>")
+	if(control)
+		boutput(user, "<span class='alert'>CTRLFREQ#[control]</span>")
+	if(beacon)
+		boutput(user, "<span class='alert'>BCKNFREQ#[beacon]</span>")
+	return
```
Comment (ZeWaka) [inline code/obj/item/tool/multitool.dm:84]:
```suggestion

```
useless

File: code/obj/item/tool/multitool.dm [hunk 2]
```diff
@@ -26,3 +26,59 @@
...
+		if (hasvar(target, "beacon_freq"))
+			beacon = target:beacon_freq
+	else if (hasvar(target, "radio_connection"))
+		var/datum/radio_frequency/radiofreq = target:radio_connection
+		frequency = radiofreq.frequency
+	else if (hasvar(target, "frequency"))
+		if(isnum(target:frequency)|istext(target:frequency))
+			frequency = target:frequency
+	//We'll do lockers safely since nothing else seems to store the frequency exactly like this
+	else if (istype(target,/obj/storage/secure))
```
Comment (ZeWaka) [inline code/obj/item/tool/multitool.dm:72]:
```suggestion
	else if (istype(target, /obj/storage/secure))
```

File: code/obj/item/tool/multitool.dm [hunk 3]
```diff
@@ -26,3 +26,59 @@
...
+		frequency = target:freq
+	else if (hasvar(target, "control_freq"))
+		control = target:control_freq
+		if (hasvar(target, "beacon_freq"))
+			beacon = target:beacon_freq
+	else if (hasvar(target, "radio_connection"))
+		var/datum/radio_frequency/radiofreq = target:radio_connection
+		frequency = radiofreq.frequency
+	else if (hasvar(target, "frequency"))
+		if(isnum(target:frequency)|istext(target:frequency))
```
Comment (ZeWaka) [inline code/obj/item/tool/multitool.dm:69]: i don't think you mean | 😸 
```suggestion
		if(isnum(target:frequency) || istext(target:frequency))
```

**Review by ZeWaka — APPROVED** [review]

**Comment by witheld9** [comment]
Hey thanks! Sorry for weirdness this is my first time actually writing DM code!

**Pull request merged**